askama_shared/src/generator.rs

```diff
@@ -1106,9 +1129,64 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
                 self.visit_method_call(buf, obj, method, args)?
             }
             Expr::RustMacro(name, args) => self.visit_rust_macro(buf, name, args),
+            Expr::Let(ref variant, ref params, ref val) => self.visit_let(buf, variant, params, val)?,
         })
     }
 
+    fn visit_let(
+        &mut self,
+        buf: &mut Buffer,
+        variant: &MatchVariant<'a>,
+        params: &Option<MatchParameters<'a>>,
+        val: &Expr<'a>,
+    ) -> Result<DisplayWrap, CompileError> {
+        self.visit_match_variant(buf, variant);
+        if let Some(params) = params {
+            match params {
+                MatchParameters::Simple(params) => {
+                    if !params.is_empty() {
+                        buf.write("(");
+                        for (i, param) in params.iter().enumerate() {
+                            if let MatchParameter::Name(p) = *param {
+                                self.locals.insert_with_default(p);
+                            }
+                            if i > 0 {
+                                buf.write(", ");
+                            }
+                            self.visit_match_param(buf, param);
+                        }
+                        buf.write(")");
+                    }
+                }
+                MatchParameters::Named(params) => {
+                    buf.write("{");
+                    for (i, param) in params.iter().enumerate() {
+                        if let Some(MatchParameter::Name(p)) = param.1 {
+                            let p = normalize_identifier(p);
+                            self.locals.insert_with_default(p);
+                        } else {
+                            self.locals.insert_with_default(param.0);
+                        }
+
+                        if i > 0 {
+                            buf.write(", ");
+                        }
+                        buf.write(param.0);
+                        if let Some(param) = &param.1 {
+                            buf.write(":");
+                            self.visit_match_param(buf, param);
+                        }
+                    }
+                    buf.write("}");
+                }
+            }
```

Comment:
Just giving it a quick glance, so I might have overlooked something. But can't all this be replaced with `self.visit_match_params(buf, params);`? :)

Reply:
Thank you for your review!

Oops, yes, you're right.